index.js

```diff
@@ -1 +1,84 @@
-const app = "I don't do much."
+var cake = {
+  name: "German Chocolate Cake",
+  ingredients: ["eggs", "flour", "oil", "chocolate", "sugar", "butter"],
+  topping: "coconut frosting",
+  bakeTemp: "425 degrees",
+  bakeTime: "45 minutes",
+  customer: "Tommy",
+  decorate: function(updateFunction) {
+    var status = "Decorating with " + this.topping + ". Ready to eat soon!"
+    updateFunction(status)
+    setTimeout(function() {
+      updateFunction(serve.apply(this, ["Happy Eating!", this.customer]))
+    }.bind(this), 2000)
+  }
+}
+
+var pie = {
+  name: "Apple Pie",
+  ingredients: ["apples", "flour", "eggs", "butter", "sugar"],
+  topping: "streusel",
+  bakeTemp: "350 degrees",
+  bakeTime: "75 minutes",
+  customer: "Tammy"
+}
+
+function makeCake() {
+  var updateCakeStatus = updateStatus.bind(this)
+  updateCakeStatus("Prep")
+  mix.call(cake, updateCakeStatus)
+}
+
+function makePie() {
+  var updatePieStatus = updateStatus.bind(this)
+  updatePieStatus("Prep")
+  pie.decorate = cake.decorate.bind(pie)
+  mix.call(pie, updatePieStatus)
+}
+
+function updateStatus(statusText) {
+  this.getElementsByClassName("status")[0].innerText = statusText
+}
+
+function mix(updateFunction) {
+  var status = "Mixing " + this.ingredients.join(", ")
+  setTimeout(function() {
+    bake.call(this, updateFunction)
+  }.bind(this), 2000)
+  updateFunction(status)
+}
+
+function bake(updateFunction) {
+  var status = "Baking at " + this.bakeTemp + " for " + this.bakeTime
+  setTimeout(function() {
```

Comment:
Could we show students these `setTimeout` calls using arrow functions in the solution?